sklearn/cluster/_birch.py

```diff
@@ -521,6 +525,13 @@ def fit(self, X, y=None):
         self
             Fitted estimator.
         """
+        if self.copy != "deprecated":
+            warnings.warn(
+                "`copy` was deprecated in 1.6 and will be removed in 1.8 since it "
+                "has no effect internally. Simply leave this parameter to its default "
+                "value to avoid this warning.",
+                FutureWarning,
+            )
```

Comment:
It will only be triggered by `fit`. We should place it in `_fit` so that it's also triggered by `partial_fit`. We need to be careful though and only trigger it for the first call to partial_fit.